FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -382,83 +387,110 @@ private async Task<List<PatientWrapper>> GetToBeProcessedPatientsAsync(Cancellat
-                                }
+                                await _metadataStore.UpdatePatientVersionsAsync(
+                                _jobInfo.QueueType,
+                                processingJobResult.ProcessedPatientVersion,
+                                cancellationToken);
+                            }
+                            catch (RequestFailedException ex)
+                            {
+                                _logger.LogError(ex, "Failed to update patient versions from metadata table.");
+                                throw new MetadataStoreException("Failed to update patient versions from metadata table.", ex);
                             }
                         }
                     }
-                    else if (latestJobInfo.Status == JobStatus.Failed)
-                    {
-                        _logger.LogInformation("The processing job is failed.");
-                        throw new RetriableJobException("The processing job is failed.");
-                    }
-                    else if (latestJobInfo.Status == JobStatus.Cancelled)
-                    {
-                        _logger.LogInformation("Operation cancelled by customer.");
-                        throw new OperationCanceledException("Operation cancelled by customer.");
-                    }
-
-                    completedJobIds.Add(latestJobInfo.Id);
                 }
+                else if (firstRunningJobInfo.Status == JobStatus.Failed)
+                {
+                    _logger.LogInformation("The processing job is failed.");
+                    throw new RetriableJobException("The processing job is failed.");
+                }
+                else if (firstRunningJobInfo.Status == JobStatus.Cancelled)
+                {
+                    _logger.LogInformation("Operation cancelled by customer.");
+                    throw new OperationCanceledException("Operation cancelled by customer.");
+                }
+
+                UpdateRunningJobStatus(firstRunningJobId);
+                progress.Report(JsonConvert.SerializeObject(_result));
             }
 
-            if (completedJobIds.Count > 0)
+            _logger.LogInformation($"Orchestrator job {_jobInfo.Id} finished checking the first running job {firstRunningJobId} status, the job {firstRunningJobId} is {((JobStatus)firstRunningJobInfo.Status).ToString("D")}.");
+        }
+
+        private long GetFirstRunningJobId()
+        {
+            return _inputData.JobVersion switch
             {
-                _result.RunningJobIds.ExceptWith(completedJobIds);
-                progress.Report(JsonConvert.SerializeObject(_result));
+                SupportedJobVersion.V1 or SupportedJobVersion.V2 => _result.RunningJobIds.Min(),
+                SupportedJobVersion.V3 => _result.SequnceIdToJobIdMap[_result.CompletedJobCount],
```

Comment:
How do we handle version conversion from V2 to V3, where field of SequnceIdToJobIdMap is empty for V2 running jobs.

Reply:
Yes, for V2, the SequnceIdToJobIdMap is empty, so runningJobIds is used. and the version of one job is fixed, it will not change, only the next new orchestrator job will change the job version from V2 to V3.